src/HeckeTypes.jl

```diff
@@ -331,6 +332,7 @@ mutable struct SRow{T, S} # S <: AbstractVector{T}
       end
     end
     r = new{elem_type(R), typeof(S)}(R, S, p)
+    @assert r isa sparse_row_type(R)
```

Comment:
This is a bit of a paranoia check that I inserted to make sure I got everything right (and it was helpful, it caught at least one bug :-)). We could now
1. keep it
2. remove it
3. turn it into a type assertion

Reply:
Just keep it